pkg/providers/aws/provider_redis.go

```diff
@@ -470,6 +471,16 @@ func buildElasticacheUpdateStrategy(elasticacheConfig *elasticache.CreateReplica
 		ec.SnapshotRetentionLimit = elasticacheConfig.SnapshotRetentionLimit
 		updateFound = true
 	}
+	for _, i := range cacheClusters {
+		if elasticacheConfig.PreferredMaintenanceWindow != nil && *elasticacheConfig.PreferredMaintenanceWindow != *i.PreferredMaintenanceWindow {
+			ec.PreferredMaintenanceWindow = elasticacheConfig.PreferredMaintenanceWindow
+			updateFound = true
+		}
+		if elasticacheConfig.SnapshotWindow != nil && *elasticacheConfig.SnapshotWindow != *i.SnapshotWindow {
```

Comment:
Can we make `i` to be more descriptive, something like `foundCacheCluster` 
Also I know it is not directly related to this PR, but can we update `ec` to be more descriptive too.